backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/impl/HUShipperTransportationBL.java

```diff
@@ -327,4 +346,59 @@ public I_M_ShipperTransportation getCommonM_ShipperTransportationOrNull(final Co
 		final I_M_ShippingPackage firstPackage = shippingPackages.iterator().next();
 		return firstPackage.getM_ShipperTransportation();
 	}
+
+	@NonNull
+	public I_M_ShipperTransportation getById(@NonNull final ShipperTransportationId shipperTransportationId)
+	{
+		return load(shipperTransportationId, I_M_ShipperTransportation.class);
+	}
+
+	@NonNull
+	public ShipperTransportationId addTrackingCodesForInOutWithoutHU(@NonNull final AddTrackingCodesForInOutWithoutHUReq req)
+	{
+		final I_M_InOut inOut = inOutDAO.getById(req.getInOutId());
+
+		final WarehouseId warehouseId =  WarehouseId.ofRepoId(inOut.getM_Warehouse_ID());
+
+		final I_M_Warehouse warehouse = warehouseDAO.getById(warehouseId);
+
+		final int bPartnerLocationId = warehouse.getC_BPartner_Location_ID();
+
+		final BPartnerLocationInfo bPartnerLocationInfo = bPartnerLocationInfoRepository.getById(bPartnerLocationId);
+
+		final CreateShipperTransportationRequest createShipperTransportationRequest = CreateShipperTransportationRequest
+				.builder()
+				.shipperId(req.getShipperId())
+				.bPartnerLocationId(bPartnerLocationInfo.getId())
+				.orgId(OrgId.ofRepoId(inOut.getAD_Org_ID()))
+				.shipDate(req.getShipDate())
+				.build();
+
+		final ShipperTransportationId shipperTransportationId = createHUShipperTransportation(createShipperTransportationRequest);
+
+		final CreatePackagesForInOutRequest createPackagesForInOutRequest = CreatePackagesForInOutRequest.builder()
+				.inOut(InterfaceWrapperHelper.create(inOut, de.metas.inout.model.I_M_InOut.class))
+				.trackingNumbers(req.getTrackingCodes())
+				.processed(true)//mark the M_Package records as processed
+				.build();
+
+		addInOutWithoutHUToShipperTransportation(shipperTransportationId, ImmutableList.of(createPackagesForInOutRequest));
+
+		return shipperTransportationId;
+	}
+
+	private ShipperTransportationId createHUShipperTransportation(@NonNull final CreateShipperTransportationRequest request)
+	{
+		final I_M_ShipperTransportation shipperTransportation = newInstance(I_M_ShipperTransportation.class);
+
+		shipperTransportation.setAD_Org_ID(request.getOrgId().getRepoId());
+		shipperTransportation.setM_Shipper_ID(request.getShipperId().getRepoId());
+		shipperTransportation.setShipper_BPartner_ID(request.getBPartnerLocationId().getBpartnerId().getRepoId());
+		shipperTransportation.setShipper_Location_ID(request.getBPartnerLocationId().getRepoId());
+		shipperTransportation.setDateDoc(Timestamp.valueOf(request.getShipDate()));
+
+		saveRecord(shipperTransportation);
```

Comment:
introduce and use IShipperTransportationDAO.save(I_M_ShipperTransportation)

Reply:
why not ShipperTransportationRepository spring component?

Reply:
even better :)